Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -839,6 +839,27 @@ extension RouteController: Router {
 extension RouteController: InternalRouter { }
 
 extension RouteController: ReroutingControllerDelegate {
+    func rerouteControllerWantsSwitchToAlternative(_ rerouteController: RerouteController, response: RouteResponse, options: RouteOptions) {
+        guard let newMainRoute = response.routes?.first else {
+            return
+        }
+        delegate?.router(self,
+                         willTakeAlternativeRoute: newMainRoute,
+                         at: location)
+        updateRoute(with: IndexedRouteResponse(routeResponse: response,
+                                               routeIndex: 0),
+                    routeOptions: options,
+                    isProactive: false,
+                    completion: { [weak self] success in
+            guard let self = self else { return }
+            if success {
+                self.delegate?.router(self, didTakeAlternativeRouteAt: self.location)
```

Comment:
Should we capture location used in `willTakeAlternativeRoute`?

Reply:
I think it makes sense to report actual location here. Thus user may track that between `willTake` and `didTake` he travelled `from ` -> `to`.

Reply:
Should we document this behavior? As one may assume that those locations are the same and i.e. use location as a key to match "will" with "did".

Reply:
Not actual anymore.